sql-plugin/src/main/java/com/nvidia/spark/rapids/GpuColumnVectorFromBuffer.java

```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020, NVIDIA CORPORATION.
+ * Copyright (c) 2020-2021, NVIDIA CORPORATION.
```

Comment:
Can we try and find a way to get rid of one of `GpuColumnVectorFromBuffer` or `GpuPackedTableColumn`? If feels odd that we would have both when they are almost exactly the same thing.

Reply:
I struggled with this at first as well, but they each serve an important and distinct function.  The sole purpose of `GpuPackedTableColumn` is to convey a buffer of data to the next stage without manifesting a `Table` or any `ColumnVector` instances, specifically so we can send these buffers to the GPU shuffle.

`GpuColumnVectorFromBuffer` is a real GPU `ColumnarBatch` and can be operated on as if it were a normal `ColumnarBatch` containing `GpuColumnVector`, just that it came from a contiguous table.  That requires manifesting the table and column instances to create it.